guppylang/std/builtins.py

```diff
@@ -597,7 +597,7 @@ def __iter__(self: array[L, n] @ owned) -> "SizedIter[ArrayIter[L, n], n]":
         return SizedIter(ArrayIter(self, 0))
 
     @guppy.custom(CopyInoutCompiler(), ArrayCopyChecker())
-    def copy(self: array[L, n]) -> array[L, n]: ...
+    def copy(self: array[T, n]) -> array[T, n]: ...
```

Comment:
The type variable `T` has a `copyable=True` requirement which better matches the actual behaviour enforcced by `ArrayCopyChecker`.

In a sense, the only purpose of `ArrayCopyChecker` now is to give a better error message if the array elements are not implicitly copyable